File: backend/src/main/kotlin/fr/gouv/dgampa/rapportnav/infrastructure/api/bff/model/v2/MissionActionDataOutput.kt
package fr.gouv.dgampa.rapportnav.infrastructure.api.bff.model.v2

import fr.gouv.dgampa.rapportnav.domain.entities.mission.nav.control.ControlAdministrativeEntity
import fr.gouv.dgampa.rapportnav.domain.entities.mission.nav.control.ControlGensDeMerEntity
import fr.gouv.dgampa.rapportnav.domain.entities.mission.nav.control.ControlNavigationEntity
import fr.gouv.dgampa.rapportnav.domain.entities.mission.nav.control.ControlSecurityEntity
import java.time.Instant

open class MissionActionDataOutput(
    open val startDateTimeUtc: Instant?,
    open val endDateTimeUtc: Instant? = null,
    open val controlSecurity: ControlSecurityEntity? = null,
    open val controlGensDeMer: ControlGensDeMerEntity? = null,
    open val controlNavigation: ControlNavigationEntity? = null,
    open val controlAdministrative: ControlAdministrativeEntity? = null,
)

File: frontend/src/v2/features/common/types/mission-action-data-output.ts
import {
  ControlAdministrative,
  ControlGensDeMer,
  ControlNavigation,
  ControlSecurity
} from '@common/types/control-types'

export interface MissionActionDataOutput {
  startDateTimeUtc: string
  endDateTimeUtc: string
  controlSecurity: ControlSecurity
  controlGensDeMer: ControlGensDeMer
  controlNavigation: ControlNavigation
  controlAdministrative: ControlAdministrative
}
